### PR TITLE
PR for review

### PR DESCRIPTION
**Pull request by Pocoder**

### CODE REVIEW

**Comment by fhoda** [comment]
I'll be going over this for the next few days and leaving comments.

**Comment by fhoda** [comment]
Thank you for opening the PR. Generally things are looking good. Most changes that I will suggest are either highly recommended or nit-picky, but help standardize things and may make it a little easier for the end user to understand the code and package in general.

I still need to look in more detail at the hook, sensors, and tasks code, but here is what i've have so far for other parts.


### Example DAGs

The following are changes we highly recommend based on best practices.

1. https://github.com/Toloka/airflow-provider-toloka/blob/cd7cc8bb755453c6ae2ff7a87445bc27e965e7f1/toloka_provider/example_dags/text_classification.py#L12 For `start_date` please use a static date and avoid using `days_ago()`

2. Move the `custom_xcom_backend` to an `include/` folder. https://github.com/Toloka/airflow-provider-toloka/blob/cd7cc8bb755453c6ae2ff7a87445bc27e965e7f1/toloka_provider/example_dags/custom_xcom_backend.py

---

The following are more nit-picky but there is potential value in the suggested changes

3. https://github.com/Toloka/airflow-provider-toloka/blob/cd7cc8bb755453c6ae2ff7a87445bc27e965e7f1/toloka_provider/example_dags/text_classification.py#L109-L112 Instead of assigning them to variables on line `109`. It would be less code to just pass each `dataset['*_task']` to the respective functions and more directly show users the dependency between the tasks in the dag code.

```
    tasks = prepare_tasks(dataset['main_tasks'])
    exam_tasks = prepare_exam_tasks(dataset['exam_tasks'])
    honeypots = prepare_honeypots(dataset['honeypots'])
```

4. General comments and/or an addition of a `doc_md` to the DAG for a description of the tasks and overall goal of the DAG


### Readme
Highly recommended

1. Add description for the hooks, sensors, and tasks so that users can understand what the provider package offers without needing to go through the code.


### Other
Nice to have
 
1. https://github.com/Toloka/airflow-provider-toloka/blob/cd7cc8bb755453c6ae2ff7a87445bc27e965e7f1/setup.py#L33 Add Trove classifiers for Apache Airflow `Framework :: Apache Airflow`

**Comment by Pocoder** [comment]
Thank you for your comments. I added more documentation as you recommended.

**Comment by fhoda** [comment]
Appreciate the changes! I have one last request to wrap up the code review...

https://github.com/Toloka/airflow-provider-toloka/blob/409f4ef7b1a786c4343c9e380385eeddb9a45f5d/toloka_provider/sensors/toloka.py#L55

It's not recommend that operators return other operators, but you can extend the `BaseSensorOperator` instead. Here are some docs on it https://airflow.apache.org/docs/apache-airflow/stable/howto/custom-operator.html#sensors

**Comment by Pocoder** [comment]
Sorry for delay, I faced some not obvious bugs with sensor. Also I've changed TolokaHook extras a little bit.

File: toloka_provider/hooks/toloka.py [hunk 1]
```diff
@@ -25,29 +29,63 @@ def __init__(self, toloka_conn_id: str = default_conn_name, *args, **kwargs) ->
...
         self.get_conn()
 
     def get_conn(self) -> TolokaClient:
-        """Function that initiates a new Toloka connection with token"""
+        """Function that creates a new TolokaClient with token and returns it"""
         if not self.client:
             self.log.debug('Creating toloka client for conn_id: %s', self.toloka_conn_id)
 
             if not self.toloka_conn_id:
                 raise AirflowException('Failed to create toloka client. no toloka_conn_id provided')
```
Comment (fhoda) [inline toloka_provider/hooks/toloka.py]:
```suggestion
                raise AirflowException('Failed to create toloka client. No toloka_conn_id provided')
```

**Comment by fhoda** [comment]
> Sorry for delay, I faced some not obvious bugs with sensor. Also I've changed TolokaHook extras a little bit.

No worries, everything looks good!

**Comment by Pocoder** [comment]
Done.

**Review by fhoda — APPROVED** [review]

**Pull request merged**